### PR TITLE
PadConv2D

### PR DESCRIPTION
**Pull request by cofri**
This PR introduces a new `PadConv2D` layer that is the combination of a padding operation and a Conv2D layer. Different padding types can be set:

- "valid": corresponds to the standard "valid" convolution in Keras Conv2D (i.e. no padding). The output shape is smaller than the input shape.
- "same": corresponds to the standard zero padding in Keras Conv2D. The output shape is equal to the input shape.
- "constant": equivalent to "same", i.e. zero padding.
- "symmetric"/"reflect": mirror padding as in `tf.pad()`. The output shape is equal to the input shape.
- "circular": circular padding. The output shape is equal to the input shape.

Moreover, some bugs are fixed and unit tests are added.

### CODE REVIEW

**Comment by thib-s** [comment]
As PadConv2D is not lipschitz constrained, we need to find a way to make sure that users are aware of it. Should we move it to a separate module for instance ?

File: deel/lip/normalizers.py [hunk 1]
```diff
@@ -96,14 +96,16 @@ def _power_iteration(w, u, eps=DEFAULT_EPS_SPECTRAL):
...
 
     """
-    # build _u and _v
+    # build _u and _v (_v is size of _u@tf.transpose(w), will be set on the first body
+    # iteration)
     _u = u
-    _v = tf.math.l2_normalize(_u @ tf.transpose(w))
-    # create a fake old_w that does'nt pass the loop condition
-    # it won't affect computation as the firt action done in the loop overwrite it.
+    _v = tf.zeros(u.shape[:-1] + (w.shape[0],))
```
Comment (thib-s) [inline deel/lip/normalizers.py:102]: Not very important but the dtype is missing (as w.dtype is not always tf.float32)

**Review by thib-s — APPROVED** [review]
Maybe ’unconstrained_layers’ could be a better name for ’layer_utils.py’ otherwise seems good to me

**Review by thib-s — APPROVED** [review]

**Review by franckma31 — APPROVED** [review]

File: deel/lip/unconstrained_layers.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
+import tensorflow as tf
+
```
Comment (thib-s) [inline deel/lip/unconstrained_layers.py:2]:
```suggestion
from tensorflow.keras.utils import register_keras_serializable
```

File: deel/lip/unconstrained_layers.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+from .layers import Condensable
+from .utils import _padding_circular
+
+
```
Comment (thib-s) [inline deel/lip/unconstrained_layers.py:6]:
```suggestion
@register_keras_serializable("deel-lip", "PadConv2D")
```

**Review by thib-s — COMMENTED** [review]
added missing serializable registration

File: deel/lip/unconstrained_layers.py (hunk 2 above)
Comment (cofri) [inline deel/lip/unconstrained_layers.py:2]:
```suggestion
from tensorflow.keras.utils import register_keras_serializable
```
